internal/node/services/lib/keyvalue.go

```diff
@@ -0,0 +1,324 @@
+package lib
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/nats-io/nats.go"
+	agentapi "github.com/synadia-io/nex/internal/agent-api"
+)
+
+const kvServiceMethodGet = "get"
+const kvServiceMethodSet = "set"
+const kvServiceMethodDelete = "delete"
+const kvServiceMethodKeys = "keys"
+
+type KeyValueService struct {
+	log *slog.Logger
+	nc  *nats.Conn
+}
+
+func NewKeyValueService(nc *nats.Conn, log *slog.Logger) (*KeyValueService, error) {
+	kv := &KeyValueService{
+		log: log,
+		nc:  nc,
+	}
+
+	err := kv.init()
+	if err != nil {
+		return nil, err
+	}
+
+	return kv, nil
+}
+
+func (k *KeyValueService) init() error {
+	return nil
+}
+
+func (k *KeyValueService) HandleRPC(msg *nats.Msg) {
+	// agentint.{vmID}.rpc.{namespace}.{service}.{method}
+	tokens := strings.Split(msg.Subject, ".")
+	service := tokens[4]
+	method := tokens[5]
+
+	switch method {
+	case kvServiceMethodGet:
+		k.handleKeyValueGet(msg)
+	case kvServiceMethodSet:
+		k.handleKeyValueSet(msg)
+	case kvServiceMethodDelete:
+		k.handleKeyValueDelete(msg)
+	case kvServiceMethodKeys:
+		k.handleKeyValueKeys(msg)
+	default:
+		k.log.Warn("Received invalid host services RPC request",
+			slog.String("service", service),
+			slog.String("method", method),
+		)
+
+		// msg.Respond()
+	}
+}
+
+func (k *KeyValueService) handleKeyValueGet(msg *nats.Msg) {
+	tokens := strings.Split(msg.Subject, ".")
+	vmID := tokens[1]
+	namespace := tokens[3]
+
+	kvStore, err := k.resolveKeyValueStore(vmID, namespace) // FIXME-- should this be worload name + namespace
```

Comment:
yes, the bucket name needs to be workload_namespace, e.g. `echoservice_default`. This is what will allow the same workload deployed on 2 different nex nodes to access the same bucket